File: app/routes/api_v2/schemas/conversion.py
"""API v2 şemaları — Dönüşüm (conversion) hunisi."""

from __future__ import annotations

from pydantic import BaseModel


class ConversionFunnel(BaseModel):
    visitors: int
    engaged: int
    clicked: int
    demo: int
    signups_landing: int
    signups_direct: int
    signups_total: int
    paid_total: int
    paid_landing: int
    rate_visitor_engaged: float
    rate_engaged_demo: float
    rate_visitor_signup: float
    rate_signup_paid: float
    rate_visitor_paid: float


class ConversionVariantRow(BaseModel):
    slug: str
    sessions: int
    signups: int
    conversion_pct: float
    paid: int
    paid_pct: float


class ConversionResponse(BaseModel):
    days: int
    funnel: ConversionFunnel
    variants: list[ConversionVariantRow]
    has_experiment: bool
    experiment_name: str | None = None

File: web/lib/types/conversion.ts
// Dönüşüm (conversion) hunisi tipleri — süper admin.

export interface ConversionFunnel {
  visitors: number;
  engaged: number;
  clicked: number;
  demo: number;
  signups_landing: number;
  signups_direct: number;
  signups_total: number;
  paid_total: number;
  paid_landing: number;
  rate_visitor_engaged: number;
  rate_engaged_demo: number;
  rate_visitor_signup: number;
  rate_signup_paid: number;
  rate_visitor_paid: number;
}

export interface ConversionVariantRow {
  slug: string;
  sessions: number;
  signups: number;
  conversion_pct: number;
  paid: number;
  paid_pct: number;
}

export interface ConversionResponse {
  days: number;
  funnel: ConversionFunnel;
  variants: ConversionVariantRow[];
  has_experiment: boolean;
  experiment_name: string | null;
}
